tests/test_themes.py

```diff
@@ -0,0 +1,179 @@
+import os
+import textwrap
+import shutil
+import pytest
+
+sep = os.path.sep
+
+
+@pytest.fixture(scope='function')
+def theme_project_tmpdir(tmpdir):
+    """Copy themes-project to a temp dir, and copy demo-project content to it."""
+    themes_dir = os.path.join(os.path.dirname(__file__), 'themes-project')
+    content_dir = os.path.join(os.path.dirname(__file__), 'demo-project', 'content')
+
+    temp_dir = tmpdir.mkdir("temp").join('themes-project')
+
+    shutil.copytree(themes_dir, str(temp_dir))
+    shutil.copytree(content_dir, str(temp_dir.join('content')))
+
+    return temp_dir
+
+
+@pytest.fixture(scope='function')
+def theme_project(theme_project_tmpdir, request):
+    """Return the theme project created in a temp dir.
+
+    Could be parametrize, if request.param=False themes variables won't be set
+    """
+    try:
+        with_themes_var = request.param
+    except AttributeError:
+        with_themes_var = True
+
+    from lektor.project import Project
+
+    # Create the .lektorproject file
+    lektorfile_text = textwrap.dedent(u"""
+        [project]
+        name = Themes Project
+        {}
+    """.format("themes = blog_theme, project_theme"
+               if with_themes_var else ""))
+
+    theme_project_tmpdir.join("themes.lektorproject").write_text(
+        lektorfile_text, "utf8", ensure=True)
+    return Project.from_path(str(theme_project_tmpdir))
+
+
+@pytest.fixture(scope='function')
+def theme_env(theme_project):
+    from lektor.environment import Environment
+    return Environment(theme_project)
+
+
+@pytest.fixture(scope='function')
+def theme_pad(theme_env):
+    from lektor.db import Database
+    return Database(theme_env).new_pad()
+
+
+@pytest.fixture(scope='function')
+def theme_builder(theme_pad, tmpdir):
+    from lektor.builder import Builder
+    return Builder(theme_pad, str(tmpdir.mkdir("output")))
+
+
+@pytest.mark.parametrize(
+    'theme_project, themes', [(True, ['blog_theme', 'project_theme']),
+                              # when themes variables isn't set themes are only loaded in the env
+                              (False, [])],
+    indirect=['theme_project'])
+def test_loading_theme_variable(theme_project, themes):
+    assert theme_project.themes == themes
+
+
+@pytest.mark.parametrize('theme_project', (True, False), indirect=True)
+def test_loading_theme_path(theme_env):
+    assert [os.path.basename(p)
+            for p in theme_env.theme_paths] == ['blog_theme', 'project_theme']
+
+
+@pytest.mark.parametrize(
+    'asset_name, found_in',
+    [
+        # - themes-project/assets/dummy.txt
+        # - themes-project/themes/blog_theme/assets/dummy.txt
+        # wil be loaded from themes-project assets not from blog_theme assets
+        ('dummy.txt', 'root'),
```

Comment:
Could you add a 'root only' test here? That would make it more consistent with the other tests here, so we have at least [root, root + theme, 2 themes (no root), 1 theme (no root)] everywhere.